src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -19,30 +19,26 @@ public class ExtensionSettings {
     // Change the location to extension.yml file of the extension
     public static final String EXTENSION_DESCRIPTOR = "src/test/resources/extension.yml";
 
-    public String getExtensionname() {
-        return extensionname;
+    public ExtensionSettings(String extensionname, String hostaddress, String hostport) {
+        this.extensionname = extensionname;
+        this.hostaddress = hostaddress;
+        this.hostport = hostport;
     }
 
-    public void setExtensionname(String extensionname) {
-        this.extensionname = extensionname;
+    public ExtensionSettings() {}
```

Comment:
Without any setters (since you've removed them) this constructor will leave the fields permanently uninitialized.

Reply:
Without a default constructor, Jackson will throw an exception

Reply:
I believe you but I also suspect there's more to this story about how Jackson wants you to write classes.